src/bin/zmlicensectl

```diff
@@ -76,6 +76,85 @@ startLicenseDaemonService()
 	fi
 }
 
+exportOfflineLicenseData() {
+    startTimeStr=""
+    endTimeStr=""
+
+    # parse start and end times from arguments
+    while [[ $# -gt 0 ]]; do
+        case "$1" in
+            --exportStartTime)
+                shift
+                startTimeStr="$1"
+                ;;
+            --exportEndTime)
+                shift
+                endTimeStr="$1"
+                ;;
+            *)
+                echo "Error: Unknown argument: $1"
+                exit 1
+                ;;
+        esac
+        shift
+    done
+
+    if [[ -n "$startTimeStr" && -z "$endTimeStr" ]] || [[ -z "$startTimeStr" && -n "$endTimeStr" ]]; then
+        echo "Error: Both --exportStartTime and --exportEndTime are required, or use no arguments to default to 90 days."
+        exit 1
+    elif [[ -n "$startTimeStr" && -n "$endTimeStr" ]]; then
+        # validating date format
+        if ! date -d "$startTimeStr" >/dev/null 2>&1 || ! date -d "$endTimeStr" >/dev/null 2>&1; then
+            echo "Error: Invalid date format. Please use YYYY/MM/DD."
+            exit 1
+        fi
+
+        # converting to Epoch timestamps
+        startEpoch=$(date -d "$startTimeStr" +%s)
+        endEpoch=$(date -d "$endTimeStr" +%s)
+
+        # input validation (dates)
+        todayEpoch=$(date +%s)
+        if (( endEpoch > todayEpoch )); then
+            echo "Error: endTime cannot be after today."
+            exit 1
+        fi
+        if (( startEpoch > endEpoch )); then
+            echo "Error: startTime must be before or equal to endTime."
+            exit 1
+        fi
+    else
+        # defaults to 90 days if no dates provided
+        echo "Defaulting export to 90 days..."
+        endEpoch=$(date +%s)
+        startEpoch=$(( endEpoch - 90 * 86400 ))
+    fi
+
+    # get the license daemon server host from zmprov
+    licenseDaemonHost=$(zmprov gcf zimbraLicenseDaemonServerHost | cut -d ':' -f2 | xargs)
```

Comment:
Can we add a check here to make sure licenseDaemonHost is not empty? If empty, print an error and exit 1.

Reply:
done, thank you!